play-json-jsoniter/src/main/scala/com/evolutiongaming/jsonitertool/PlayJsonJsoniter.scala

```diff
@@ -0,0 +1,115 @@
+package com.evolutiongaming.jsonitertool
+
+import com.github.plokhotnyuk.jsoniter_scala.core._
+import play.api.libs.json._
+
+object PlayJsonJsoniter {
+
+  implicit val jsValueCodec: JsonValueCodec[JsValue] =
+    new JsonValueCodec[JsValue] {
+
+      /**
+       * The implementation was borrowed from: https://github.com/plokhotnyuk/jsoniter-scala/blob/e80d51019b39efacff9e695de97dce0c23ae9135/jsoniter-scala-benchmark/src/main/scala/io/circe/CirceJsoniter.scala
+       * and adapted to meet PlayJson criteria.
+       */
+      override def decodeValue(in: JsonReader, default: JsValue): JsValue = {
+        var b = in.nextToken
+        if (b == 'n') in.readNullOrError(default, "expected `null` value")
+        else if (b == '"') {
+          in.rollbackToken()
+          JsString(in.readString(null))
+        } else if (b == 'f' || b == 't') {
+          in.rollbackToken()
+          if (in.readBoolean()) JsTrue else JsFalse
+        } else if (b == '-' || (b >= '0' && b <= '9')) {
+          JsNumber(
+            {
+              in.rollbackToken
+              in.setMark
+              try {
+                do {
+                  b = in.nextByte
+                } while ((b >= '0' && b <= '9') || b == '-')
+              } catch {
+                case _: JsonReaderException => /* ignore end of input error */
+              } finally in.rollbackToMark
+              //PlayJson specific thing, since it uses BigDecimal to represent all numbers.
+              in.readBigDecimal(null)
```

Comment:
Let me check

Reply:
Couldn't reproduce any error with string representation of max double value. Works as expected for both scenarios.

val maxDoubleStr = "179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368"

//BigDecimal(maxDoubleStr).toDouble == Double.MaxValue

case class MaxD(max: Double)

val json = s"""{ "max":${maxDoubleStr} }"""

implicit val e = Json.reads[MaxD]
implicit val f = Json.writes[MaxD]

//parse str -> serialize to bytes(Jsoniter) -> read it back (PlayJson)
val jsValue0 = Json.parse(json)
val bts = PlayJsonJsoniter.serialize(jsValue0)
val jsValue1 = Json.parse(bts)
jsValue0 shouldEqual jsValue1

//parse str -> serialize to bytes(PlayJson) -> read it back (Jsoniter)
val jsValue0 = Json.parse(json)
val bts = Json.toBytes(jsValue0)
val jsValue1 =  PlayJsonJsoniter.deserialize(bts)
jsValue0 shouldEqual jsValue1

Reply:
If u have smth to add to that, feel free to do so.

Reply:
```
    val jsValue0 = Json.parse(json.getBytes)
    val jsValue1 = PlayJsonJsoniter.deserialize(json.getBytes)
    jsValue0 shouldEqual jsValue1
```
fails with 
```
[info]   com.github.plokhotnyuk.jsoniter_scala.core.JsonReaderException: value exceeds limit for number of digits, offset: 0x0000013b, buf:
[info] +----------+-------------------------------------------------+------------------+
[info] |          |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f | 0123456789abcdef |
[info] +----------+-------------------------------------------------+------------------+
[info] | 00000110 | 37 32 33 31 36 38 37 33 38 31 37 37 31 38 30 39 | 7231687381771809 |
[info] | 00000120 | 31 39 32 39 39 38 38 31 32 35 30 34 30 34 30 32 | 1929988125040402 |
[info] | 00000130 | 36 31 38 34 31 32 34 38 35 38 33 36 38 20 7d    | 6184124858368 }  |
[info] +----------+-------------------------------------------------+------------------+
```

Reply:
You can use [an overloaded method](https://github.com/plokhotnyuk/jsoniter-scala/blob/master/jsoniter-scala-core/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/core/JsonReader.scala#L388) with additional parameters for the math context and limits scale and mantissa digits.

Reply:
```
val s = JsonParserSettings.settings.bigDecimalParseSettings
JsNumber(in.readBigDecimal(null, s.mathContext, s.scaleLimit, s.digitsLimit))
```
@plokhotnyuk would it be safe to use it like this? 

https://github.com/playframework/play-json/blob/f0cc0858a4aa0cbf45304d1422e729a7d663b946/play-json/jvm/src/main/scala/play/api/libs/json/JsonParserSettings.scala#L61

Reply:
It looks like in order to be able to parse 310 characters numbers by Jsoniter without a failure we need to pass ```s.digitsLimit+1``` 
```
val s = JsonParserSettings.settings.bigDecimalParseSettings
JsNumber(in.readBigDecimal(null, s.mathContext, s.scaleLimit, s.digitsLimit+1))
```

Reply:
@haghard on my machine it works fine with just `s.digitsLimit`, thing is that 310 in play is because there's a negative value of Double.MaxValue, so it's not 310 digits, but 309 digits and a `-` sign.
so no need to `s.digitsLimit+1`

Reply:
Will show u tomorrow that I mean and we will decide then.